### PR TITLE
Refactor lock type to reflect linear state progression.

### PR DESCRIPTION
**Pull request by ExternalReality**
The lock type's tracking of unit state is made linear with this change.

### CODE REVIEW

**Comment by ExternalReality** [comment]
[Lock Type and States Refactor](https://trello.com/c/OdbzV0IM/441-lock-type-and-states-refactor)

**Comment by ExternalReality** [comment]
!!build!!

**Comment by ExternalReality** [comment]
!!build!!

**Comment by ExternalReality** [comment]
!!build!!

**Comment by ExternalReality** [comment]
!!build!!

File: core/model/upgradeseries.go [hunk 1]
```diff
@@ -7,103 +7,29 @@ import (
...
-type UnitSeriesUpgradeStatus string
+// Machine upgrade series status will be removed
+type MachineSeriesUpgradeStatus = UpgradeSeriesStatus
 
 const (
-	UnitNotStarted UnitSeriesUpgradeStatus = "NotStarted"
-	UnitStarted    UnitSeriesUpgradeStatus = "Started"
-	UnitErrored    UnitSeriesUpgradeStatus = "Errored"
-	UnitCompleted  UnitSeriesUpgradeStatus = "Completed"
+	NotStarted       UpgradeSeriesStatus = "not started"
```
Comment (manadart) [inline core/model/upgradeseries.go]: I would be inclined to keep these suffixed with "UpgradeSeries" to disambiguate them from other constants that this package may export in the future.

File: apiserver/common/upgradeseries.go [hunk 2]
```diff
@@ -142,27 +142,27 @@ func (u *UpgradeSeriesAPI) WatchUpgradeSeriesNotifications(args params.Entities)
 // UpgradeSeriesPrepareStatus returns the current preparation status of an upgrading
 // unit. If no series upgrade is in progress an error is returned instead.
 func (u *UpgradeSeriesAPI) UpgradeSeriesPrepareStatus(args params.Entities) (params.UpgradeSeriesStatusResults, error) {
```
Comment (manadart) [inline apiserver/common/upgradeseries.go]: These two methods can become one.

File: apiserver/common/upgradeseries.go [hunk 3]
```diff
@@ -142,27 +142,27 @@ func (u *UpgradeSeriesAPI) WatchUpgradeSeriesNotifications(args params.Entities)
...
 // UpgradeSeriesCompleteStatus returns the current completion status of upgrading
 // unit. If no series upgrade is in progress an error is returned instead.
 func (u *UpgradeSeriesAPI) UpgradeSeriesCompleteStatus(args params.Entities) (params.UpgradeSeriesStatusResults, error) {
-	return u.upgradeSeriesStatus(args, model.CompleteStatus)
+	return u.upgradeSeriesStatus(args)
 }
 
 // SetUpgradeSeriesPrepareStatus sets the upgrade series status of the unit.
 // If no upgrade is in progress an error is returned instead.
 func (u *UpgradeSeriesAPI) SetUpgradeSeriesPrepareStatus(args params.SetUpgradeSeriesStatusParams) (params.ErrorResults, error) {
```
Comment (manadart) [inline apiserver/common/upgradeseries.go]: Same with these two.

File: state/machine.go [hunk 4]
```diff
@@ -2340,24 +2327,11 @@ func (m *Machine) isMachineUpgradeSeriesStatusSet(status model.UpgradeSeriesStat
...
-	// TODO (manadart 2018-08-08): Remove when lock structure modified.
-	t, s := status.ToOldStatus()
-	var field string
-	switch t {
-	case model.PrepareStatus:
-		field = "prepare-status"
-	case model.CompleteStatus:
-		field = "complete-status"
-	}
+	field := "prepare-status"
```
Comment (manadart) [inline state/machine.go]: Just "status" ?

File: state/machine_test.go [hunk 5]
```diff
@@ -2736,34 +2738,39 @@ func (s *MachineSuite) TestMachineSeriesUpgradeStatusTranslatesCorrectly(c *gc.C
 	err := s.machine.CreateUpgradeSeriesLock([]string{unit0.Name()}, "cosmic")
 	c.Assert(err, jc.ErrorIsNil)
 
-	sts, err := s.machine.MachineUpgradeSeriesStatus()
+	status, err := s.machine.MachineUpgradeSeriesStatus()
```
Comment (manadart) [inline state/machine_test.go]: I tend to avoid naming conflicts with imports. It can make for subtle bugs.

**Comment by ExternalReality** [comment]
!!build!!

File: state/machine.go [hunk 6]
```diff
@@ -2384,9 +2358,8 @@ func completeUpgradeSeriesTxnOps(machineDocID string, units []unitStatus) []txn.
 			C:  machineUpgradeSeriesLocksC,
 			Id: machineDocID,
 			Assert: bson.D{{"$and", []bson.D{
-				//{{"prepare-status", model.MachineSeriesUpgradeComplete}}, //[TODO]externalreality: re-enable this check
-				{{"complete-status", model.MachineSeriesUpgradeNotStarted}}}}},
-			Update: bson.D{{"$set", bson.D{{"complete-units", units}}}},
+				{{"prepare-status", model.PrepareCompleted}}}}},
```
Comment (jameinel) [inline state/machine.go]: If the previous one is changing to 'field := "status"' then I think this one needs to change as well.

**Comment by ExternalReality** [comment]
$$MERGE$$

**Pull request merged**